### ISSUE TITLE
Naga generates SPIR-V for WGSL `fallthrough` that Naga cannot read

### ISSUE THREAD

**Issue by blakej11** [gfx-rs/naga#1099]
The following WGSL code:

```
fn fiddle(x: i32) -> i32 {
    var r: i32;

    r = 42;

    switch (x) {
        case 1: { fallthrough; }
        case 0: { r = 0; }
    }

    return (r);
}
```

validates successfully, and if it's converted to SPIR-V using Naga, [edit] that is accepted by `spirv-val`. However, if that SPIR-V is fed to Naga as input, it fails to validate, with the following error message:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ControlFlowGraphCycle(16)', cli/src/main.rs:147:64
```

This is similar to, but distinct from, https://github.com/gfx-rs/naga/issues/1088. 

I'm hitting this in a wgpu/wasm program because of https://github.com/gfx-rs/wgpu/pull/1618. Presumably once that's
 fixed this won't be an urgent issue for me.

**Comment by kvark** [comment]
Another spv-in CFG issue, cc @jimblandy who wanted to look at rewriting the CFG processing

**Comment by jimblandy** [comment]
Edited title and description to make it clear that this is a SPIR-V input bug, not a SPIR-V generation bug. The original title and description made it sound as if the SPIR-V was invalid, but the generated SPIR-V is valid SPIR-V - it's just that Naga can't read it.

**Comment by JCapucho** [comment]
Fixed by #1294
